File: apps/scraper/query/sources/library_hours.py
"""ASU Library hours for this week, every location."""

from __future__ import annotations

from scraper.core.types import QuerySource
from scraper.sources.library_hours import SOURCE as HOURS


def to_url(params: dict[str, str]) -> str:
    """The hours page. It takes no parameters."""
    return HOURS.url


QUERY = QuerySource(
    key="library_hours",
    description="Fetch this week's opening hours for every ASU library.",
    params=(),
    to_url=to_url,
    needs_js=HOURS.needs_js,
    extractor=HOURS.extractor,
)

File: apps/engine/src/runtime/tools/knowledge/search/library_hours.rs
//! This week's opening hours at every ASU library.

use super::{LiveSource, Param};

/// The ASU Library hours page.
pub struct LibraryHours;

impl LiveSource for LibraryHours {
    fn key(&self) -> &'static str {
        "library_hours"
    }

    fn hint(&self) -> &'static str {
        "library opening hours"
    }

    fn label(&self) -> &'static str {
        "ASU Library Hours"
    }

    fn category(&self) -> &'static str {
        "library"
    }

    fn params(&self) -> &'static [Param] {
        &[]
    }
}
